pkg/resource/managed_reconciler.go

```diff
@@ -406,6 +410,23 @@ func (r *ManagedReconciler) Reconcile(req reconcile.Request) (reconcile.Result,
 			return reconcile.Result{RequeueAfter: r.shortWait}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
 		}
 
+		if !managed.IsConditionTrue(v1alpha1.TypeReferencesResolved) {
+			if err := r.managed.ResolveReferences(ctx, managed); err != nil {
+				// update the status according to the type of the err
+				switch err.(type) {
```

Comment:
I thought about this, but avoided thinking it might be too much. I implement the corresponding error interface.

Reply:
>When one commands ResolveReferences the expectation is that the command is either executed successfully or has some errors

I think this is right. Personally, I'd go for `GetUnresolvedReferences` since IMO we don't actually resolve but wait for the resources to be ready. But that's highly opinionated, I can see your case fitting as well.

> I think the rational behind using an error is similar to Get case in most cloud provider go APIs

Actually, I'd compare `Find` or `List` instead of `Get`. In `Get`, you assume the resource exists, so, getting non-existent resource is a problem. But it should be fine to return empty array to `List` or `Find`.

> I like the idea of returning an array of unresolved references as the error message though. Do you think that would be useful?

Yes, I'd love to know that detail if my resource gets stuck at that stage for a long time for some reason.

Reply:
I have change the `NotReadyError`, so that it accepts a list of items as an input, and uses that in `Error` function.
https://github.com/crossplaneio/crossplane-runtime/pull/46/files#diff-45d446315c96683e664c463ff0674883R59-R61

Reply:
> Yes, I'd love to know that detail if my resource gets stuck at that stage for a long time for some reason.

@muvaf I changed the code to return a detailed status for the references. See the last commit